### ISSUE TITLE
Is there a model template of redock_default2018 model?

### ISSUE THREAD

**Issue by lkfo415579** [gnina/scripts#41]
Hello, Thanks for your contribution to the SBDD community, you guys are doing almost ten years about AI + SBDD project, I have read papers of those wonderful jobs.

Recently, I want to reproduce the default models you guys had made but I could not find any related config or script to reproduce it. I want to understand the details behind those model, hopefully you guys can provide more information.

Below is the list of models I want to reproduce.
crossdock_default2018 
default2017 
dense 
general_default2018 
redock_default2018

Thanks again.

**Comment by dkoes** [comment]
Look here: https://github.com/gnina/models/tree/master/crossdocked_paper

**Comment by lkfo415579** [comment]
> Look here: https://github.com/gnina/models/tree/master/crossdocked_paper

I have read this page but when I look at the default2018.model config, this setting is not for training **redock dataset**, because the huber loss is not **turn on**. And I think it is an important detail setting allow model to be pose sensitive.

I have 2 questions.
1. Is there a true config specific for training redock dataset?
2. When I use Hinge-like loss and huber loss to train redock dataset, the performance of the model is not as good as a model trains purely on Positive Samples. Therefore I am wondering which part of details I am missing? Gnina-CNN model can achieve very good performance while using positive + negative samples.

`layer {
  name: "rmsd"
  type: "AffinityLoss"
  bottom: "affinity_output"
  bottom: "affinity"
  top: "rmsd"
  affinity_loss_param {
    scale: 0.1
    gap: 0
    pseudohuber: false
    delta: 4
    penalty: 0
    ranklossmult: 0
    ranklossneg: 0    
  }
}`

**Comment by francoep** [comment]
The model configs used in that location are the model configs for all the models that we trained. 

When training the models are you using our provided training scripts (train.py being the main one) and types files (instructions to download the types files are here-- https://github.com/gnina/models/tree/master/data/CrossDocked2020)?

Our general training for the redocked model was 
```
train.py -m <modelfile> -p <typesfile prefix> -n <fold number> -s <random seed> -t 1000 -i <some large number like 900000000> -o <output prefix> --dynamic --lr_policy fixed --step_when 200 --percent_reduced 3.82 --skip_full
```
This sets up the training script to handle our built-in learning rate lowering and stops training once the model hasn't improved.

**Comment by lkfo415579** [comment]
Thanks for the quick reply.

Currently I am using my **own model** to train your types files(redock dataset), meanwhile analyzing the training procedure of gnina-cnn model.

You said your general training is almost as same as the config on above, I got that. But what about those hyperparameters? (ranking loss, pseudo-Huber, gap) can they further improve the performance of the model?

And why you guys scale affinity loss by 0.1? but not the pose loss? In my opinion because the portion of affinity data is only 34% (less than classification samples), you should scale the pose loss instead of affinity loss.

I am not quite familiar with caffee haha..

**Comment by dkoes** [comment]
All the hyperparameters were determined as part of a large hyperparameter sweep.  The scaling of affinity loss is useful because the magnitude of that loss is significantly higher than the pose loss.

**Comment by lkfo415579** [comment]
Thanks for your reply, it does help me a lot.

**Comment by Kerro-junior** [comment]
I find the following information in README in crossdock2020 models :

```
These model files are blank slates for training your own models.
In order to properly train a new model the following needs to change within the model file:

LIGCACHE_FILE -- needs to be changed to a *_lig.molcache2 file or delete the line.
RECCACHE_FILE -- needs to be changed to a *_rec.molcache2 file or delete the line.
Then you can proceed with train.py in gnina/scripts.
```
Does this mean that if I have to train on different molcacheI2 files, I'd better delete these lines in the model file, and assign them in the train.py?

But I cannot find the cache2 file position in train.py, there is only types file that could be assigned.
`( -p PREFIX, --prefix PREFIX     Prefix for training/test files: <prefix>[train|test][num].types)`

**Comment by francoep** [comment]
That would be because you need to specify the cache in the caffe .model file.  For best results you should have a cache that contains the information in the .types files that you will be training.

If no cache is specified in the .model file, then libmolgrid will read the files specified in the .types file from disk (very slow if you have many training examples)
